### PR TITLE
Refactor number of cells magic number

### PR DESCRIPTION
**Pull request by damienrg**
Done to remove code duplication and improve readability.

### CODE REVIEW

File: cartographer/mapping/probability_values.cc [hunk 1]
```diff
@@ -23,12 +23,14 @@ namespace mapping {
...
+
 // 0 is unknown, [1, 32767] maps to [lower_bound, upper_bound].
 float SlowValueToBoundedFloat(const uint16 value, const uint16 unknown_value,
                               const float unknown_result,
                               const float lower_bound,
                               const float upper_bound) {
-  CHECK_LE(value, 32767);
+  CHECK_LT(value, kNumberOfCells);
   if (value == unknown_value) return unknown_result;
   const float kScale = (upper_bound - lower_bound) / 32766.f;
```
Comment (MichaelGrupp) [inline cartographer/mapping/probability_values.cc]: What about this number?

Comment (damienrg) [reply cartographer/mapping/probability_values.cc]: The condition was "less equal" and it is now "less than".

Comment (MichaelGrupp) [reply cartographer/mapping/probability_values.cc]: I meant 32766.f in this line: `const float kScale = (upper_bound - lower_bound) / 32766.f`

Comment (damienrg) [reply cartographer/mapping/probability_values.cc]: Done.

Comment (damienrg) [reply cartographer/mapping/probability_values.cc]: While removing this, I was thinking to change the variable to `kNumberOfValues`.
WDYT?

**Comment by pifon2a** [comment]
@wally-the-cartographer merge

**Comment by wally-the-cartographer** [comment]
Merge requested by authorized user pifon2a. Merge queue now has a length of 1.

**Comment by wally-the-cartographer** [comment]
Refusing to merge, since the combined checks state is failure

**Comment by damienrg** [comment]
@pifon2a, I have rebased on master and made 2 minor updates.

**Comment by pifon2a** [comment]
Thank you!

**Pull request merged**